### PR TITLE
fix(rest) : modified attachment info in response to the moderation request rest api

### PR DESCRIPTION
**Pull request by keerthi-bl**
Closes : #2497 

### Description

Please describe your issue in few words here.

#### How to reproduce

Describe the bug and list the steps you used when the issue occurred.

#### Screenshots ( if applicable )

image

### CODE REVIEW

**Comment by rudra-superrr** [comment]
Resolve the merge conflicts

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/moderationrequest/ModerationRequestController.java [hunk 1]
```diff
@@ -116,26 +124,36 @@ public ResponseEntity<CollectionModel<ModerationRequest>> getModerationRequests(
         return getModerationResponseEntity(pageable, request, allDetails, modRequestsWithPageData);
     }
 
-    @Operation(
-            summary = "Get a single moderation request.",
-            description = "Get a single moderation request by id.",
-            tags = {"Moderation Requests"}
-    )
+    @Operation(summary = "Get a single moderation request.", description = "Get a single moderation request by id.", tags = {
+            "Moderation Requests" })
```
Comment (GMishx) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/moderationrequest/ModerationRequestController.java:128]: Please revert irrelevant changes.

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/moderationrequest/ModerationRequestController.java [hunk 2]
```diff
@@ -116,26 +124,36 @@ public ResponseEntity<CollectionModel<ModerationRequest>> getModerationRequests(
...
     @RequestMapping(value = MODERATION_REQUEST_URL + "/{id}", method = RequestMethod.GET)
-    public ResponseEntity<HalResource<ModerationRequest>> getModerationRequestById(
-            @Parameter(description = "The id of the moderation request to be retrieved.")
-            @PathVariable String id
-    ) throws TException {
+    public ResponseEntity getModerationRequestById(
```
Comment (GMishx) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/moderationrequest/ModerationRequestController.java:130]: Any reason to remove the object type from `ResponseEntity`? It causes wrong Spring documents to be generated.

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/moderationrequest/ModerationRequestController.java (hunk 1 above)
Comment (keerthi-bl) [reply rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/moderationrequest/ModerationRequestController.java]: reverted

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/moderationrequest/ModerationRequestController.java (hunk 2 above)
Comment (keerthi-bl) [reply rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/moderationrequest/ModerationRequestController.java]: Added proper Object type.

**Comment by rudra-superrr** [comment]
Testing was successful.
image

**Comment by rudra-superrr** [comment]
One thing that I have found is, when a user updates a project then MR is created but if the same user updates some other field of the same project then it overwrites the previous moderation request.

**Comment by GMishx** [comment]
> One thing that I have found is, when a user updates a project then MR is created but if the same user updates some other field of the same project then it overwrites the previous moderation request.

Can you please open an issue for the same? We need to rework on the Moderation Request with REST API. With the Liferay UI, we had the liberty to check when an MR will be created and how to treat it. But with REST API, currently there is no way to check the same and we have to revert to this behavior.

Please check https://github.com/eclipse-sw360/sw360/pull/2405#issuecomment-2426291315 as well.

**Comment by rudra-superrr** [comment]
> > One thing that I have found is, when a user updates a project then MR is created but if the same user updates some other field of the same project then it overwrites the previous moderation request.
> 
> Can you please open an issue for the same? We need to rework on the Moderation Request with REST API. With the Liferay UI, we had the liberty to check when an MR will be created and how to treat it. But with REST API, currently there is no way to check the same and we have to revert to this behavior.
> 
> Please check [#2405 (comment)](https://github.com/eclipse-sw360/sw360/pull/2405#issuecomment-2426291315) as well.

But this issue is happening because of the changes made in this PR.
Please have a look into this @keerthi-bl

**Comment by rudra-superrr** [comment]
> > One thing that I have found is, when a user updates a project then MR is created but if the same user updates some other field of the same project then it overwrites the previous moderation request.
> 
> Are you sure about this finding @rudra-superrr ? This PR is modifying only GET endpoint. Ideally it should not effect any modification.

We both also discussed regarding the same point but while testing issue is coming with this branch.

**Comment by GMishx** [comment]
Okay,. Keeping this branch on hold then.

**Comment by GMishx** [comment]
@rudra-superrr I just tested and confirmed by @keerthi-bl as well, the issue exists with `main` branch not with this PR.

Created issue #2828 for the same.

**Review by GMishx — APPROVED** [review]
Changes looks good.

**Pull request merged**